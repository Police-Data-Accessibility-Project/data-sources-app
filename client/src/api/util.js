export function isCachedResponseValid({
	cacheTime,
	intervalBeforeInvalidation = 1000 * 60 * 2,
	currentTime = new Date().getTime(),
}) {
	const cacheAge = currentTime - (cacheTime ?? 0);

	return cacheAge < intervalBeforeInvalidation;
}
