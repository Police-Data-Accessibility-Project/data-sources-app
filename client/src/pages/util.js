export const STATIC_VIEW_UI_SHAPE = [
	{
		header: "Data Type",
		records: [
			{
				title: "Record type",
				key: "record_type",
				classNames:
					"mt-1 py-[.125rem] px-3 rounded-full bg-brand-wine/10 dark:bg-brand-wine dark:text-white w-fit small",
				component: "button",
			},
			{ title: "Description", key: "description" },
			{ title: "Tags", key: "tags" },
		],
	},
	{
		header: "Access & format",
		records: [
			{ title: "Source URL", key: "source_url", component: "a" },
			{ title: "ReadMe URL", key: "readme_url" },
			{ title: "Access Type", key: "access_type" },
			{
				title: "Record Formats",
				key: "record_format",
				classNames:
					"mt-1 py-[.125rem] px-3 rounded-full bg-slate-200 dark:bg-slate-600 w-fit small",
			},
			{ title: "Detail Level", key: "detail_level" },
			{ title: "Size", key: "size" },
			{ title: "Access Notes", key: "access_notes" },
			// records_not_online to be hidden
			// { title: "Records Not Online", key: "records_not_online" },
		],
	},
	{
		header: "Agency",
		records: [
			{
				title: "Name",
				key: "agency_name",
				component: "button",
				classNames:
					"decoration-solid decoration-[6%] text-brand-gold  underline-offset-[5%] hover:brightness-85 small",
			},
			{ title: "State", key: "state_iso" },
			{ title: "County", key: "county_name" },
			{ title: "Municipality", key: "municipality" },
			{ title: "Agency Type", key: "agency_type" },
			{ title: "Jurisdiction Type", key: "jurisdiction_type" },
		],
	},
	{
		header: "Provenance",
		records: [
			{ title: "Agency Supplied", key: "agency_supplied" },
			{ title: "Supplying Entity", key: "supplying_entity" },
			{ title: "Agency Originated", key: "agency_originated" },
			{ title: "Originating Entity", key: "originating_entity" },
		],
	},
	{
		header: "Coverage & retention",
		records: [
			{
				title: "Coverage Start Date",
				key: "coverage_start",
				isDate: true,
			},
			{ title: "Coverage End Date", key: "coverage_end", isDate: true },
			{
				title: "Source Last Updated",
				key: "source_last_updated",
				isDate: true,
			},
			{ title: "Update Frequency", key: "update_frequency" },
			{ title: "Update Method", key: "update_method" },
			{ title: "Retention Schedule", key: "retention_schedule" },
			{
				title: "Number of Records Available",
				key: "number_of_records_available",
			},
		],
	},
	{
		header: "Data Source Meta",
		records: [
			{ title: "Scraper URL", key: "scraper_url" },
			{ title: "Created", key: "data_source_created", isDate: true },
			{ title: "Agency ID", key: "agency_id" },
			{ title: "Data Source ID", key: "data_source_id" },
		],
	},
];
