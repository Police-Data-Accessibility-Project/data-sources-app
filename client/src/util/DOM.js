export function isDescendantOf(child, parent) {
	if (!child || !parent) return false;
	if (child === parent) return true;
	return isDescendantOf(child.parentNode, parent);
}
