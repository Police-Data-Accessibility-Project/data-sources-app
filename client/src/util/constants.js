import { FOOTER_LINK_ICONS } from 'pdap-design-system';

export const ALL_LOCATION_TYPES = ['locality', 'county', 'state', 'federal'];

export const NAV_LINKS = [
	{
		path: '/',
		text: 'Search',
		method: 'route',
	},
	{
		href: 'https://pdap.io/data',
		text: 'Data',
		method: 'href',
	},
	{
		href: 'https://pdap.io/community',
		text: 'Community',
		method: 'href',
	},
	{
		href: 'https://pdap.io/about',
		text: 'About',
		method: 'href',
	},
	{
		href: 'https://pdap.io/donate',
		text: 'Donate',
		method: 'href',
	},
	{
		href: 'https://docs.pdap.io/',
		text: 'Docs',
		method: 'href',
	},
];

export const FOOTER_LINKS = [
	{
		href: 'https://github.com/orgs/Police-Data-Accessibility-Project',
		text: 'Github',
		icon: FOOTER_LINK_ICONS.GITHUB,
	},
	{
		href: 'https://discord.gg/wMqex8nKZJ',
		text: 'Discord',
		icon: FOOTER_LINK_ICONS.DISCORD,
	},
	{
		href: 'https://www.linkedin.com/company/pdap',
		text: 'LinkedIn',
		icon: FOOTER_LINK_ICONS.LINKEDIN,
	},
	{
		href: 'https://pdap.io/jobs',
		text: 'Jobs',
		icon: FOOTER_LINK_ICONS.JOBS,
	},
	{
		href: 'https://newsletter.pdap.io/',
		text: 'Newsletter',
		icon: FOOTER_LINK_ICONS.NEWSLETTER,
	},
	{
		href: 'https://docs.pdap.io/',
		text: 'Docs',
		icon: FOOTER_LINK_ICONS.DOCS,
	},
];

export const STATES_TO_ABBREVIATIONS = new Map([
	['Alabama', 'AL'],
	['Alaska', 'AK'],
	['American Samoa', 'AS'],
	['Arizona', 'AZ'],
	['Arkansas', 'AR'],
	['Armed Forces Americas', 'AA'],
	['Armed Forces Europe', 'AE'],
	['Armed Forces Pacific', 'AP'],
	['California', 'CA'],
	['Colorado', 'CO'],
	['Connecticut', 'CT'],
	['Delaware', 'DE'],
	['District Of Columbia', 'DC'],
	['Florida', 'FL'],
	['Georgia', 'GA'],
	['Guam', 'GU'],
	['Hawaii', 'HI'],
	['Idaho', 'ID'],
	['Illinois', 'IL'],
	['Indiana', 'IN'],
	['Iowa', 'IA'],
	['Kansas', 'KS'],
	['Kentucky', 'KY'],
	['Louisiana', 'LA'],
	['Maine', 'ME'],
	['Marshall Islands', 'MH'],
	['Maryland', 'MD'],
	['Massachusetts', 'MA'],
	['Michigan', 'MI'],
	['Minnesota', 'MN'],
	['Mississippi', 'MS'],
	['Missouri', 'MO'],
	['Montana', 'MT'],
	['Nebraska', 'NE'],
	['Nevada', 'NV'],
	['New Hampshire', 'NH'],
	['New Jersey', 'NJ'],
	['New Mexico', 'NM'],
	['New York', 'NY'],
	['North Carolina', 'NC'],
	['North Dakota', 'ND'],
	['Northern Mariana Islands', 'NP'],
	['Ohio', 'OH'],
	['Oklahoma', 'OK'],
	['Oregon', 'OR'],
	['Pennsylvania', 'PA'],
	['Puerto Rico', 'PR'],
	['Rhode Island', 'RI'],
	['South Carolina', 'SC'],
	['South Dakota', 'SD'],
	['Tennessee', 'TN'],
	['Texas', 'TX'],
	['US Virgin Islands', 'VI'],
	['Utah', 'UT'],
	['Vermont', 'VT'],
	['Virginia', 'VA'],
	['Washington', 'WA'],
	['West Virginia', 'WV'],
	['Wisconsin', 'WI'],
	['Wyoming', 'WY'],
]);

export const ABBREVIATIONS_TO_STATES = new Map([
	['AL', 'Alabama'],
	['AK', 'Alaska'],
	['AS', 'American Samoa'],
	['AZ', 'Arizona'],
	['AR', 'Arkansas'],
	['AA', 'Armed Forces Americas'],
	['AE', 'Armed Forces Europe'],
	['AP', 'Armed Forces Pacific'],
	['CA', 'California'],
	['CO', 'Colorado'],
	['CT', 'Connecticut'],
	['DE', 'Delaware'],
	['DC', 'District Of Columbia'],
	['FL', 'Florida'],
	['GA', 'Georgia'],
	['GU', 'Guam'],
	['HI', 'Hawaii'],
	['ID', 'Idaho'],
	['IL', 'Illinois'],
	['IN', 'Indiana'],
	['IA', 'Iowa'],
	['KS', 'Kansas'],
	['KY', 'Kentucky'],
	['LA', 'Louisiana'],
	['ME', 'Maine'],
	['MH', 'Marshall Islands'],
	['MD', 'Maryland'],
	['MA', 'Massachusetts'],
	['MI', 'Michigan'],
	['MN', 'Minnesota'],
	['MS', 'Mississippi'],
	['MO', 'Missouri'],
	['MT', 'Montana'],
	['NE', 'Nebraska'],
	['NV', 'Nevada'],
	['NH', 'New Hampshire'],
	['NJ', 'New Jersey'],
	['NM', 'New Mexico'],
	['NY', 'New York'],
	['NC', 'North Carolina'],
	['ND', 'North Dakota'],
	['NP', 'Northern Mariana Islands'],
	['OH', 'Ohio'],
	['OK', 'Oklahoma'],
	['OR', 'Oregon'],
	['PA', 'Pennsylvania'],
	['PR', 'Puerto Rico'],
	['RI', 'Rhode Island'],
	['SC', 'South Carolina'],
	['SD', 'South Dakota'],
	['TN', 'Tennessee'],
	['TX', 'Texas'],
	['VI', 'US Virgin Islands'],
	['UT', 'Utah'],
	['VT', 'Vermont'],
	['VA', 'Virginia'],
	['WA', 'Washington'],
	['WV', 'West Virginia'],
	['WI', 'Wisconsin'],
	['WY', 'Wyoming'],
]);
