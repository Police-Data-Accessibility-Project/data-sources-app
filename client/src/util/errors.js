export class DataLoaderErrorPassThrough extends Error {
	constructor(message) {
		super(message);
		this.name = 'DataLoaderErrorPassThrough';
	}
}
