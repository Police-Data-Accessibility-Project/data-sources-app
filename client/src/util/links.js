export const links = [
	{
		path: '/',
		text: 'Search',
		method: 'route',
	},
	{
		href: 'https://pdap.io/data',
		text: 'Data',
		method: 'href',
	},
	{
		href: 'https://pdap.io/community',
		text: 'Community',
		method: 'href',
	},
	{
		href: 'https://pdap.io/about',
		text: 'About',
		method: 'href',
	},
	{
		href: 'https://pdap.io/donate',
		text: 'Donate',
		method: 'href',
	},
	{
		href: 'https://docs.pdap.io/',
		text: 'Docs',
		method: 'href',
	},
];
