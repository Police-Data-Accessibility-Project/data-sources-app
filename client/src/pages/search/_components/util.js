import { STATES_TO_ABBREVIATIONS } from '@/util/constants';

// TODO: replace with common util
export function formatText(location) {
	return (
		location.locality_name +
		', ' +
		STATES_TO_ABBREVIATIONS.get(location.state_name)
	);
}
