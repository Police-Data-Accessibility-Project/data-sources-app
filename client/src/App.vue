<template>
	<AuthWrapper>
		<Header :logo-image-src="lockup" />
		<router-view />
		<Footer :logo-image-src="acronym" />
	</AuthWrapper>
</template>

<script>
import { Footer, Header } from 'pdap-design-system';
import AuthWrapper from './components/AuthWrapper.vue';
import acronym from 'pdap-design-system/images/acronym.svg';
import lockup from 'pdap-design-system/images/lockup.svg';

import { links } from './util/links';

export default {
	name: 'App',
	components: {
		AuthWrapper,
		Header,
		Footer,
	},
	provide: {
		navLinks: [...links],
		footerLinks: [...links],
	},
	data() {
		return {
			acronym,
			lockup,
		};
	},
};
</script>

<style>
#app {
	margin: 0;
}

main {
	min-height: calc(100% - 80px - 500px);
}
</style>
