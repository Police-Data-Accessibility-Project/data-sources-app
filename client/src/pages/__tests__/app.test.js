import { mount, RouterLinkStub, RouterViewStub } from '@vue/test-utils';
import App from '../../App.vue';
import { beforeEach, describe, expect, it } from 'vitest';
import { links } from '../../util/links';
import { createTestingPinia } from '@pinia/testing';

let wrapper;

describe('App', () => {
	beforeEach(() => {
		wrapper = mount(App, {
			global: {
				plugins: [createTestingPinia()],
				provide: {
					navLinks: links,
					footerLinks: links,
				},
				stubs: {
					RouterLink: RouterLinkStub,
					RouterView: RouterViewStub,
				},
			},
		});
	});

	it('renders the App', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('provides the correct navLinks and footerLinks data', () => {
		expect(wrapper.vm.$options.provide.navLinks).toEqual(links);
		expect(wrapper.vm.$options.provide.footerLinks).toEqual(links);
	});
});
