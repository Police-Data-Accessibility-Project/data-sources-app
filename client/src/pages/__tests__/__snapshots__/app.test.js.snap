// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders the App 1`] = `
<div class="h-full w-full" id="wrapper">
  <header class="pdap-header">
    <a class="logo">
      <img alt="Police Data Accessibility Project Logo" src="/node_modules/pdap-design-system/dist/images/lockup.svg" width="250">
    </a>
    <div aria-controls="nav" class="pdap-nav-open-button" role="button">
      <svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z" />
      </svg>
    </div>
    <nav aria-expanded="false" class="pdap-nav" id="nav" style="top: 0px;">
      <li class="pdap-nav-link-container">
        <!---->
        <a active-class="pdap-nav-link-current" class="pdap-nav-link" exact-active-class="pdap-nav-link-current-exact">Search</a>
      </li>
      <li class="pdap-nav-link-container">
        <a class="pdap-nav-link" href="https://pdap.io/data" referrerpolicy="no-referrer" target="_blank">Data</a>
        <!---->
      </li>
      <li class="pdap-nav-link-container">
        <a class="pdap-nav-link" href="https://pdap.io/community" referrerpolicy="no-referrer" target="_blank">Community</a>
        <!---->
      </li>
      <li class="pdap-nav-link-container">
        <a class="pdap-nav-link" href="https://pdap.io/about" referrerpolicy="no-referrer" target="_blank">About</a>
        <!---->
      </li>
      <li class="pdap-nav-link-container">
        <a class="pdap-nav-link" href="https://pdap.io/donate" referrerpolicy="no-referrer" target="_blank">Donate</a>
        <!---->
      </li>
      <li class="pdap-nav-link-container">
        <a class="pdap-nav-link" href="https://docs.pdap.io/" referrerpolicy="no-referrer" target="_blank">Docs</a>
        <!---->
      </li>
    </nav>
  </header>
  <!---->
  <footer class="pdap-footer">
    <div class="pdap-flex-container pdap-flex-container-center">
      <ul class="pdap-footer-social-links">
        <li class="pdap-footer-link-container">
          <!---->
          <a active-class="pdap-footer-social-link-current" class="pdap-footer-social-link" exact-active-class="pdap-footer-social-link-current-exact">Search</a>
        </li>
        <li class="pdap-footer-link-container">
          <a class="pdap-footer-social-link" href="https://pdap.io/data" referrerpolicy="no-referrer" target="_blank">Data</a>
          <!---->
        </li>
        <li class="pdap-footer-link-container">
          <a class="pdap-footer-social-link" href="https://pdap.io/community" referrerpolicy="no-referrer" target="_blank">Community</a>
          <!---->
        </li>
        <li class="pdap-footer-link-container">
          <a class="pdap-footer-social-link" href="https://pdap.io/about" referrerpolicy="no-referrer" target="_blank">About</a>
          <!---->
        </li>
        <li class="pdap-footer-link-container">
          <a class="pdap-footer-social-link" href="https://pdap.io/donate" referrerpolicy="no-referrer" target="_blank">Donate</a>
          <!---->
        </li>
        <li class="pdap-footer-link-container">
          <a class="pdap-footer-social-link" href="https://docs.pdap.io/" referrerpolicy="no-referrer" target="_blank">Docs</a>
          <!---->
        </li>
      </ul>
      <p class="pdap-footer-copyright"> Â© 2024 Police Data Accessibility Project
        <br>
      </p>
      <p class="pdap-footer-copyright"> PDAP is a non-profit. EIN: 85-4207132. <a alt="Privacy Policy" href="https://docs.pdap.io/meta/policy/pdap-privacy-policy" target="_blank">
          <br> Privacy Policy <i class="fa fa-external-link"></i>
        </a>
        <br>
        <a href="mailto:contact@pdap.io">contact@pdap.io</a>
      </p>
      <div class="pdap-footer-widget-links">
        <a href="https://www.guidestar.org/profile/85-4207132" target="_blank">
          <img src="https://widgets.guidestar.org/gximage2?o=9973356&l=v4">
        </a>
        <a class="pdap-footer-logo">
          <img alt="Police Data Accessibility Project Logo" src="/node_modules/pdap-design-system/dist/images/acronym.svg" width="250">
        </a>
      </div>
    </div>
  </footer>
</div>
`;
